### PR TITLE
Fix: guard PadOfConst against SmallVector overflow on large tosa pads

### PR DESCRIPTION
**Pull request by SamNour**

### CODE REVIEW

File: test/mlir/onnx/onnx_constprop_flags.mlir [hunk 1]
```diff
@@ -38,3 +39,57 @@ func.func @test_cast_f16_i16() -> tensor<6xi16> {
...
+// Large expansion: never folds with bound≤2.
+// This represents the class of over-padded weight constants that triggered the
+// original SmallVector overflow crash.
+func.func @test_pad_doesnt_propagate_expansion_bound() -> tensor<3x10xf32> {
+  %data = onnx.Constant dense<[[1.0, 1.2], [2.3, 3.4], [4.5, 5.7]]> : tensor<3x2xf32>
+  %pads = onnx.Constant dense<[0, 8, 0, 0]> : tensor<4xi64>
+  %non = "onnx.NoValue"() {value} : () -> none
+  %1 = "onnx.Pad"(%data, %pads, %non, %non) {mode = "constant"} : (tensor<3x2xf32>, tensor<4xi64>, none, none) -> tensor<3x10xf32>
+  onnx.Return %1 : tensor<3x10xf32>
+}
```
Comment (jorickert) [inline test/mlir/onnx/onnx_constprop_flags.mlir:91]: I don't get how we could have got an overflow her. SmallTensor can easily hold 3x10 elements?

Comment (SamNour) [reply test/mlir/onnx/onnx_constprop_flags.mlir:91]: The purpose here is to test the `expansionBound` constraint introduced to the `constProb` pattern [here](https://github.com/Xilinx/onnx-mlir/pull/631/changes#diff-289e9e6ac2bdcf995d2d655d62a5e2ed7b4277e2c67c2325837d36f29a7d1ae8R1181).

We can validate the expansion bound via:
```
sum_of_operand_bytes × bound  >=  output_bytes
```
taking this test as an example:  the checking at both bound 1 and 2 are blocked
```
Check at bound=1: 56 × 1 = 56 >= 120 // blocked
Check at bound=2: 56 × 2 = 112 >=120 // blocked
```
This is where it comes to bound checking. 

Regarding the overflow - setting aside that the `expansionBound` constraint was not registered - the pass runs with expansion bound default = -1 (no limit), so we folded without any expansion bound checking. This is now guarded by checking if the output type satisfies the max smallVector capacity.

Comment (jorickert) [reply test/mlir/onnx/onnx_constprop_flags.mlir:91]: I think adding a test with very large dims would be nice, to test the SmallVector overflow fix

**Review by jorickert — APPROVED** [review]
